File: 3_Longest_Substring_Without_Repeating_Characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        if(s.length() == 0) return 0;
        int max = 1;
        for(int i = 0; i < s.length(); i++){
            int sub = i + 1;
            while(sub < s.length() && !s.substring(i, sub).contains(Character.toString(s.charAt(sub)))){
                    sub++;
            }
            if(sub - i > max){
                max = sub - i;
            }
        }
        return max;
    }
}

File: 3_Longest_Substring_Without_Repeating_Characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        m = {}
        start = 0
        res = 0
        curr = 0
        for i in range(len(s)):
            if s[i] not in m or m[s[i]] < start:
                curr += 1
                res = max(res, curr)
            else:
                start = m[s[i]] + 1
                curr = i - m[s[i]]
            m[s[i]] = i
        return res